raiden-ts/src/services/epics.ts

```diff
@@ -429,15 +442,133 @@ export const monitorUdcBalanceEpic = (
   {}: Observable<RaidenAction>,
   {}: Observable<RaidenState>,
   { address, latest$, userDepositContract }: RaidenEpicDeps,
-): Observable<udcDeposited> =>
+): Observable<udcDeposit.success> =>
   latest$.pipe(
     pluckDistinct('state', 'blockNumber'),
     // it's seems ugly to call on each block, but UserDepositContract doesn't expose deposits as
     // events, and ethers actually do that to monitor token balances, so it's equivalent
     exhaustMap(() => userDepositContract.functions.effectiveBalance(address) as Promise<UInt<32>>),
-    distinctUntilChanged((x, y) => y.eq(x)),
-    map(udcDeposited),
+    withLatestFrom(latest$),
+    filter(([balance, { udcBalance }]) => !udcBalance.eq(balance)),
```

Comment:
We don't. This isn't an event, just the current balance, but since the UDC doesn't have an event for this, it's harder to confirm the balance. An option would be to check `confirmationBlocks` in the past, and call it the confirmed balance. I'll implement.